Added Work Item Revision Replay Migration.

hkoestin opened the pull request:
This work item migration context will not directly migrate over the last revision of the source work item, but it will migrate over all revisions from the source item, thus resulting in the very same history and revision graph on the new target system.

MrHinsh commented:
@hkoestin this is awesome work and I would love to have @rfennell have a look as well.

- Can you add a switch to WorkItemRevisionReplayContext for enable/disable revision migration (we will move to this as the primary migration context and retire the existing one with this feature.)

- Can you please revise the code above and then ping me on martin@nkdagility.com to make sure I see it

-MrHinsh

MrHinsh commented:
@hkoestin I want to merge this, but need you to also update the version number to 7.1.0

hkoestin commented:
@MrHinsh I updated the git-version to 7.1.0 now

Related to your questions in #16 , I added a validation for the configuration now. the tool only starts, if either the normal work item migration context is in the config, or the work item revision replay context.
